samza-core/src/main/java/org/apache/samza/operators/MessageStreamImpl.java

```diff
@@ -183,14 +185,15 @@ public void init(Config config, TaskContext taskContext) {
   }
 
   @Override
-  public MessageStream<M> merge(Collection<MessageStream<? extends M>> otherStreams) {
+  public MessageStream<M> merge(Collection<? extends MessageStream<? extends M>> otherStreams) {
     MessageStreamImpl<M> nextStream = new MessageStreamImpl<>(this.graph);
-
-    otherStreams.add(this);
-    otherStreams.forEach(other -> {
-        OperatorSpec mergeOperatorSepc =
+    List<MessageStream<M>> streamsToMerge = new ArrayList<>();
```

Comment:
Why not leverage the ArrayList constructor  (e.g. new ArrayList<>(otherStreams)) to copy the otherStreams collection and then add "this"?